x-pack/plugins/siem/public/components/open_timeline/delete_timeline_modal/index.test.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash/fp';
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
+import * as React from 'react';
+
+import { DeleteTimelineModalButton } from '.';
+import { HasIsDisabled } from '../types';
+
+describe('DeleteTimelineModal', () => {
+  const savedObjectId = 'abcd';
+
+  describe('showModalState', () => {
+    test('it disables the delete icon if deleteTimelines is not provided', () => {
+      const wrapper = mountWithIntl(
+        <DeleteTimelineModalButton savedObjectId={savedObjectId} title="Privilege Escalation" />
+      );
+
+      const props = wrapper
+        .find('[data-test-subj="delete-timeline"]')
+        .first()
+        .props() as HasIsDisabled;
```

Comment:
I wouldn't write any new specific interfaces/types from another file and then cast it for testing when we already have the interfaces and types.

For example, tracing through the component that is tied to `[data-test-subj="delete-timeline"]` looks like it is tied to the interface `EuiButtonIconProps` and you can do this:

```ts
import { EuiButtonIconProps } from '@elastic/eui';
      const props = wrapper
        .find('[data-test-subj="delete-timeline"]')
        .first()
        .props() as EuiButtonIconProps;
```

If you just want that one property to be very specific you can use Pick like so:

```ts
import { EuiButtonIconProps } from '@elastic/eui';
type HasIsDisabled = Pick<EuiButtonIconProps, 'isDisabled'>;
      const props = wrapper
        .find('[data-test-subj="delete-timeline"]')
        .first()
        .props() as EuiButtonIconProps;
```

Pros are:
* You can then delete the types you created and we have less code to maintain
* Once back in master, if Eui changes that property name they will see a type script error